hastory-server/src/Data/Hastory/Server/Handler/Entries.hs

```diff
@@ -1,12 +1,25 @@
-module Data.Hastory.Server.Handler.Entries where
+module Data.Hastory.Server.Handler.Entries
+  ( createEntryHandler
+  ) where
 
 import Data.Hastory.Server.Handler.Import
 
-createEntryHandler :: AuthCookie -> SyncRequest -> HastoryHandler NoContent
-createEntryHandler cookie syncReq =
-  withUser (unAuthCookie cookie) $ \user -> do
-    let upsertIfNotExist = upsertBy uniqueContentHash serverEntry []
-        uniqueContentHash = UniqueContentHash (serverEntryContentHash serverEntry)
-        serverEntry = toServerEntry syncReq (entityKey user)
-    _ <- runDB upsertIfNotExist
-    pure NoContent
+createEntryHandler ::
+     AuthCookie -> SyncRequest -> ServerEntryId -> HastoryHandler [Entity ServerEntry]
+createEntryHandler cookie syncReq logPosition =
+  withUser (unAuthCookie cookie) $ \user ->
+    insertNewEntries user syncReq >> fetchEntriesGreaterThan user logPosition
+
+insertNewEntries :: Entity User -> SyncRequest -> HastoryHandler [Entity ServerEntry]
+insertNewEntries user syncReq = do
+  let serverEntries = toServerEntries syncReq (entityKey user)
+  forM serverEntries $ \serverEntry -> do
+    let uniqueContentHash = UniqueContentHash (serverEntryContentHash serverEntry)
+        upsertIfNotExist = upsertBy uniqueContentHash serverEntry []
```

Comment:
A comment about the empty list would be nice.

Reply:
I left a comment, and also replaced [upsertBy](https://hackage.haskell.org/package/persistent-2.10.5.2/docs/Database-Persist-Class.html#v:upsertBy) with [upsert](https://hackage.haskell.org/package/persistent-2.10.5.2/docs/Database-Persist-Class.html#v:upsert).